### PR TITLE
fix: connect on every nwc request to ensure reconnect happens

### PR DESCRIPTION
**Pull request by rolznz**

### CODE REVIEW

File: src/webln/NostrWeblnProvider.ts [hunk 1]
```diff
@@ -340,8 +337,8 @@ export class NostrWebLNProvider implements WebLNProvider, Nip07Provider {
     );
   }
 
-  keysend(args: KeysendArgs) {
-    this.checkConnected();
+  async keysend(args: KeysendArgs) {
+    await this.checkConnected();
 
     return this.executeNip47Request<SendPaymentResponse, Nip47PayResponse>(
```
Comment (bumi) [inline src/webln/NostrWeblnProvider.ts:343]: do those need to be awaited now? 

is it good to mix async/await with returned promisses?

**Comment by bumi** [comment]
This now changes the API - which might be OK, but now calling enable is no longer required.

shouldn't we still require that? and check if `enable` was called and if so we do call the connect function

File: src/webln/NostrWeblnProvider.ts (hunk 1 above)
Comment (rolznz) [reply src/webln/NostrWeblnProvider.ts:343]: It's just the same as doing
```
const result = await ...;
return result;
```

Comment (bumi) [reply src/webln/NostrWeblnProvider.ts:344]: thx.

**Review by bumi — APPROVED** [review]

**Pull request merged**